src/Anonymization/Anonymizer/AnonymizerRegistry.php

```diff
@@ -4,134 +4,233 @@
+                    if (\preg_match('(^phar:)i', $sourceFile) === 0) {
+                        $sourceFile = \realpath($sourceFile);
+                    }
+
+                    // Require file for its content to be present in the
+                    // \get_declared_classes() function result.
+                    require_once $sourceFile;
+                }
+            }
 
-                $includedFiles[] = $sourceFile;
+            if ($found) {
+                foreach (\get_declared_classes() as $className) {
+                    $this->addAnonymizer($className, false);
+                }
             }
         }
+    }
 
-        $declared = \get_declared_classes();
-
-        foreach ($declared as $className) {
-            if ((new \ReflectionClass($className))->getAttributes(AsAnonymizer::class)) {
-                if (!\is_subclass_of($className, AbstractAnonymizer::class)) {
-                    throw new \InvalidArgumentException(\sprintf(
-                        '"%s" should extends "%s".',
-                        $className,
-                        AbstractAnonymizer::class
-                    ));
-                }
+    /**
+     * Add anonymizer definition.
+     */
+    private function addAnonymizer(string $className, bool $failOnError = false): void
+    {
+        $refClass = new \ReflectionClass($className);
 
-                $classes[$className::id()] = $className;
+        // Ignore nbn-concrete usable classes.
+        if ($refClass->isAbstract()) {
+            if ($failOnError) {
+                throw new \InvalidArgumentException(\sprintf("'%s': class is abstract.", $className));
+            }
+            return;
+        }
+        if (!$refClass->isSubclassOf(AbstractAnonymizer::class)) {
+            if ($failOnError) {
+                throw new \InvalidArgumentException(\sprintf("'%s': class does not extend '%s'.", $className, AbstractAnonymizer::class));
             }
+            return;
         }
 
-        $this->anonymizers = $classes;
-
-        return $classes;
+        $this->anonymizers[$className::id()] = $className;
     }
 
-    public function get(string $name): string
+    /**
+     * Locate installed packs.
+     */
+    private function locatePacks(): void
     {
-        if (!isset($this->getAnonymizers()[$name])) {
-            throw new \InvalidArgumentException(\sprintf("Can't find Anonymizer with name : %s, check your configuration.", $name));
-        }
+        if (\class_exists(InstalledVersions::class)) {
+
+            $ignoreList = '@(composer|doctrine|nikic|monolog|phar-io|phpunit|phpstan|psr|sebastian|symfony|twig)/*@';
 
-        return $this->getAnonymizers()[$name];
+            foreach (InstalledVersions::getInstalledPackages() as $package) {
+                if (\preg_match($ignoreList, $package)) {
+                    continue;
+                }
+
+                $directory = InstalledVersions::getInstallPath($package);
```

Comment:
with `InstalledVersions::getInstalledPackages` you could avoid to parse `composer.json` files below

Reply:
Done !